### ISSUE TITLE
ESLint: Cannot find module 'eslint-plugin-vue' Require stack

### ISSUE THREAD

**Issue by qct5566** [pnpm/pnpm#3005]
### Code to reproduce the issue:
`
ESLint: Cannot find module 'eslint-plugin-vue' Require stack: - F:\data-platform-pnpm-store\pnpm\node_modules\.pnpm\eslint@5.16.0\node_modules\eslint\lib\config\config-file.js - F:\data-platform-pnpm-store\pnpm\node_modules\.pnpm\eslint@5.16.0\node_modules\eslint\lib\config.js - F:\data-platform-pnpm-store\pnpm\node_modules\.pnpm\eslint@5.16.0\node_modules\eslint\lib\cli-engine.js - F:\data-platform-pnpm-store\pnpm\node_modules\.pnpm\eslint@5.16.0\node_modules\eslint\lib\api.js - c:\Users\Administrator\.vscode\extensions\dbaeumer.vscode-eslint-2.1.13\server\out\eslintServer.js Referenced from: F:\QCT\work\linewell\dgp\dgp-workflow\workflow\src\main\resources\templates\workflow-vue\workflow-frontend\.eslintrc.js. Please see the 'ESLint' output channel for details.
`
After using the pnpm install project dependency, I found a warning when I opened vscode. He told me that the eslint-plugin-vue was not found. However, after I checked the pnpm-store and node_modules, I found that the package existed normally

In the. npmrc file of the project, I added:
`
package-import-method= auto
shamefully-hoist = true
`
pnpm-store, I put it in  F:\data-platform-pnpm-store\pnpm. I'm windows

I tried many methods in the document and searched for issues. It seems that the problems generated by eslint-plugin were solved by using public-hoist-pattern, and public-hoist-pattern = * is equivalent to shamefully-hoist = true

I'm not sure if this is a vscode problem, because I found similar problems in webstorm. I hope the author can help me

### Additional information:

 - `node -v` prints: 12.16.2
 - Windows, macOS, or Linux?:Windows

**Comment by zkochan** [comment]
`eslint-plugin-vue` is actually hoisted by default as we hoist all packages that start with `eslint-plugin-`.

Is ESLint working when executed directly through the console?

**Comment by zaicevas** [comment]
> `eslint-plugin-vue` is actually hoisted by default as we hoist all packages that start with `eslint-plugin-`.
> 
> Is ESLint working when executed directly through the console?

I've stumbled across this issue when testing a custom ESLint plugin with `pnpm link`. Indirect `eslint-plugin-` and `eslint-config-` dependencies are being hoisted if I install a module from NPM registry or through local tarball

However, I want to test custom eslint plugin (which has more eslint plugins and configs as dependencies) through `pnpm link`. What would you recommend in this case? Is it expected that dependencies are not hoisted with `pnpm link`?

Some more details of my use case:

1. `eslint-plugin-bar` is used in `foo`
2. `eslint-plugin-bar` uses `eslint-plugin-testing-library`
2. Run `pnpm link --global` in `eslint-plugin-bar`
3. Run `pnpm link --global eslint-plugin-bar` in `foo`

`eslint-plugin-testing-library` is not hoisted and I get `ESLint: Cannot find module 'eslint-plugin-testing-library`.

**Comment by zkochan** [comment]
Try to use the ["injected"](https://pnpm.io/package_json#dependenciesmetainjected) feature. Declare `"injected": true ` for you linked plugin.

**Comment by zaicevas** [comment]
> Try to use the ["injected"](https://pnpm.io/package_json#dependenciesmetainjected) feature. Declare `"injected": true ` for you linked plugin.

Hm, I am failing to understand how this feature is of help, due to my limited knowledge of `pnpm`. Is that what you mean? ⬇️ 

```
  // foo package.json
  "dependenciesMeta": {
    "eslint-plugin-bar": {
      "injected": true
    }
  }
```

Doesn't seem to help :/

**Comment by zkochan** [comment]
yes, this should work. Is `"eslint-plugin-bar"` also in the dependencies?

**Comment by zaicevas** [comment]
> yes, this should work. Is `"eslint-plugin-bar"` also in the dependencies?

Yeah. What I did:

1. Manually add `eslint-plugin-bar` to dependencies and `dependenciesMeta` in foo `package.json`:
```
 "dependencies": {
    "eslint-plugin-bar": "0.1.0"
  },
  "dependenciesMeta": {
    "@vinted/eslint-plugin": {
      "injected": true
    }
  }
```
2. `pnpm link --global` in `eslint-plugin-bar`
3. `pnpm link --global eslint-plugin-bar` in foo
4. `pnpm build` in `eslint-plugin-bar`
5. `npx eslint file.ts` in foo

Result: `ESLint: Cannot find module 'eslint-plugin-testing-library'` 

There's no symlink of `eslint-plugin-testing-library` (and other eslint deps of `eslint-plugin-bar`) in `foo/node_modules`

**Comment by zkochan** [comment]
You need to do this:

```
 "dependencies": {
    "eslint-plugin-bar": "0.1.0"
  },
  "dependenciesMeta": {
    "eslint-plugin-bar": {
      "injected": true
    }
  }
```

**Comment by zaicevas** [comment]
> You need to do this:
> 
> ```
>  "dependencies": {
>     "eslint-plugin-bar": "0.1.0"
>   },
>   "dependenciesMeta": {
>     "eslint-plugin-bar": {
>       "injected": true
>     }
>   }
> ```

Yes, that's what I did, it doesn't help 😞  (I left a renaming mistake when I initially posted my previous comment, but that's not the issue)

**Comment by zaicevas** [comment]
It is a scoped module, if that changes anything

**Comment by zkochan** [comment]
Closing: current pnpm covers this.

An ESLint plugin has to be resolvable from the project. Declare the plugin as a dependency, or hoist it with `publicHoistPattern`. Not a pnpm defect.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
